sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -0,0 +1,185 @@
+"""Contains MSSQLEngineAdapter."""
+
+
+from __future__ import annotations
+
+import contextlib
+import typing as t
+
+import pandas as pd
+from sqlglot import exp
+
+from sqlmesh.core.engine_adapter.base import EngineAdapterWithIndexSupport
+from sqlmesh.core.engine_adapter.mixins import (
+    LogicalReplaceQueryMixin,
+    PandasNativeFetchDFSupportMixin,
+)
+from sqlmesh.core.engine_adapter.shared import DataObject, DataObjectType
+from sqlmesh.utils.errors import SQLMeshError
+
+if t.TYPE_CHECKING:
+    import pymssql
+
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter._typing import Query, QueryOrDF
+
+
+class MSSQLEngineAdapter(
+    EngineAdapterWithIndexSupport,
+    LogicalReplaceQueryMixin,
+    PandasNativeFetchDFSupportMixin,
+):
+    """Implementation of EngineAdapterWithIndexSupport for MsSql compatibility.
+
+    Args:
+        connection_factory: a callable which produces a new Database API-compliant
+            connection on every call.
+        dialect: The dialect with which this adapter is associated.
+        multithreaded: Indicates whether this adapter will be used by more than one thread.
+    """
+
+    DIALECT: str = "tsql"
+
+    def table_exists(self, table_name: TableName) -> bool:
+        """
+        Similar to Postgres, MsSql doesn't support describe so I'm using what
+        is used there and what the redshift cursor does to check if a table
+        exists. We don't use this directly in order for this to work as a base
+        class for other postgres.
+
+        Reference: https://github.com/aws/amazon-redshift-python-driver/blob/master/redshift_connector/cursor.py#L528-L553
+        """
+        table = exp.to_table(table_name)
+
+        catalog_name = table.args.get("catalog") or "master"
+        sql = (
+            exp.select("1")
+            .from_(f"{catalog_name}.information_schema.tables")
+            .where(f"table_name = '{table.alias_or_name}'")
+        )
+        database_name = table.args.get("db")
+        if database_name:
+            sql = sql.where(f"table_schema = '{database_name}'")
+
+        self.execute(sql)
+
+        result = self.cursor.fetchone()
+
+        return result[0] == 1 if result is not None else False
```

Comment:
`if result` seems more appropriate in this case since it'd else handle an empty collection.